src/Type/Constant/ConstantStringType.php

```diff
@@ -278,6 +278,16 @@ public function toArrayKey(): Type
 		return $this->arrayKeyType = is_int($offsetValue) ? new ConstantIntegerType($offsetValue) : new ConstantStringType($offsetValue);
 	}
 
+	public function isInteger(): TrinaryLogic
+	{
+		return TrinaryLogic::createMaybe();
+	}
+
+	public function isFloat(): TrinaryLogic
+	{
+		return TrinaryLogic::createMaybe();
```

Comment:
These additions are wrong. ConstantStringType can never be an integer, nor a float.